pages/voorgenomen-huwelijk/getuigen/index.tsx

```diff
@@ -155,3 +173,68 @@ export default function MultistepForm1() {
     </Surface>
   );
 }
+
+const mapWitnesses = (witnesses: Witness[]) => {
+  return witnesses
+    .filter((witness) => witness.name.length > 0 || witness.email.length > 0)
+    .map((witness) => {
+      const { name, email } = witness;
+      return {
+        name: name,
+        requester: null,
+        contact: {
+          voornaam: name,
+          emails: [{ naam: name, email: email }],
+        },
+      };
+    });
+};
+
+const WitnessFieldset = ({ index }: { index: number }) => {
+  const { t } = useTranslation(["form"]);
+  const { register, formState, watch } = useFormContext<WitnessFormData>();
+  const witnessId = useId();
+  const nameId = useId();
+  const emailId = useId();
+
+  const nameInvalid = !!formState.errors.witnesses?.[index]?.name;
+  const emailInvalid = !!formState.errors.witnesses?.[index]?.email;
+
+  return (
+    <Fieldset invalid={nameInvalid || emailInvalid}>
+      <FieldsetLegend>
+        {t("form:legal-witness")} {index}
+      </FieldsetLegend>
+      <FormField>
+        <p className="utrecht-form-field__label">
+          <FormLabel htmlFor={nameId}>{t("form:name")}</FormLabel>
+        </p>
+        {formState.errors.witnesses?.[index]?.name?.type === "required" && (
+          <FormFieldDescription invalid>{t("form:name-required")}</FormFieldDescription>
+        )}
+        <Textbox
+          id={nameId}
+          type="text"
+          autoComplete={`name ${witnessId}`}
+          invalid={nameInvalid}
+          {...register(`witnesses.${index}.name`)}
```

Comment:
if you add the `required` option with the  error message you'll want to show you don't have to create your own errors later. https://react-hook-form.com/docs/useform/register 

So 
```js
{...register(`witnesses.${index}.name`, {required: 'some error message'})}
```

Reply:
or you can just use `required: true` and keep the error message in the invalid form-field description

Reply:
It's a bit iffy here though... we don't necessarily want witnesses to be required, but rather, we want one of the fields to be required only if the related `email` or `name` field have any values. Setting required as above would result in the browser forcing a value regardless, even though the user might not want to add all witnesses.

Reply:
I thought of a few other ways to solve this. One would be to check during the render if the `name` field has any value, set `email` required and vice versa, another could be to have the input fields only appear as the user presses some sort of `Add Witness` button and allow the user to remove them. However, this seemed like the easiest to understand and implement for now

Reply:
Ooooh yes good point. Didn't think of the required only when the other is filled in as the purpose here. I agree this is the most readable solution for this usecase then! 😅